src/Util/CompositionUtil.spec.tsx

```diff
@@ -0,0 +1,116 @@
+      expect(dummyComponent.props).not.toHaveProperty(dummyField);
+
+      const injected = CompositionUtil.injectProperties({
+        component: dummyComponent,
+        onChange: dummyOnChange,
+        propName: dummyField,
+        propValue: dummyValue
+      });
+
+      expect(injected.props).toHaveProperty('onChange');
+      expect(injected.props).toHaveProperty(dummyField);
+      expect(injected.props[dummyField]).toEqual(dummyValue);
+    });
+  });
+
+  describe('handleComposition', () => {
+    it('returns default element if no composition was defined', () => {
+      dummyComposition = {};
+      const composition = CompositionUtil.handleComposition({
+        composition: dummyComposition,
+        path: dummyPath,
+        onChange: dummyOnChange,
+        propName: dummyField,
+        propValue: dummyValue,
+        defaultElement: dummyDefaultElement
+      });
+      
+      expect(composition).toBeDefined();
+      expect(composition.type).toEqual(dummyDefaultElement.type);
+      expect(composition.type).not.toEqual(dummyComponent.type);
+      expect(composition.type).not.toEqual(dummyGlobalComponent.type);
+    });
+
+    it('prefers nested options over global options', () => {
+      const composition = CompositionUtil.handleComposition({
+        composition: dummyComposition,
+        path: dummyPath,
+        onChange: dummyOnChange,
+        propName: dummyField,
+        propValue: dummyValue,
+        defaultElement: dummyDefaultElement
+      });
+
+      expect(composition).toBeDefined();
+      expect(composition.type).toEqual(dummyComponent.type);
+      expect(composition.type).not.toEqual(dummyGlobalComponent.type);
+      expect(composition.type).not.toEqual(dummyDefaultElement.type);
+    });
+
+    it('uses the global option if not nested option is defined', () => {
+      delete dummyComposition.dummyEditor.dummyField;
+      const composition = CompositionUtil.handleComposition({
+        composition: dummyComposition,
+        path: dummyPath,
+        onChange: dummyOnChange,
+        propName: dummyField,
+        propValue: dummyValue,
+        defaultElement: dummyDefaultElement
+      });
+
+      expect(composition).toBeDefined();
+      expect(composition.type).toEqual(dummyGlobalComponent.type);
+      expect(composition.type).not.toEqual(dummyComponent.type);
+      expect(composition.type).not.toEqual(dummyDefaultElement.type);
+    });
+
+    it('returns null if property is set to false', () => {
+      dummyComposition.dummyEditor.dummyField = false;
+      const composition = CompositionUtil.handleComposition({
+        composition: dummyComposition,
+        path: dummyPath,
+        onChange: dummyOnChange,
+        propName: dummyField,
+        propValue: dummyValue,
+        defaultElement: dummyDefaultElement
+      });
+
+      expect(composition).toBeDefined();
+      expect(composition).toBeFalsy();
+
+    });
+  });
+});
```

Comment:
newline please

Reply:
solved in https://github.com/geostyler/geostyler/pull/1186/commits/587b9b93e34174a44cce9a8464bcb50512687670